src/Motions/Document.ts

```diff
@@ -20,6 +21,13 @@ export class MotionDocument extends Motion {
         return MotionDocument.toLine({n: args.n === undefined ? +Infinity : args.n});
     }
 
+    static toLinePercent(args: {n: number}): Motion {
+        const obj = new MotionDocument();
+        obj.percent = args.n/100;
+        obj.line = 0;
```

Comment:
It's better we clamp the `obj.percent` between 0 and 1 here.
And we can omit assignment to `line`.

Reply:
Well, I tried this but line seems to be undefined which renders true the first conditional statement inside apply and early return. Did I miss some trick here?